src/Test/ImplicitMethodsIntegrationTest.php

```diff
@@ -106,7 +164,7 @@ public function testExplicitRequest(string $method, MiddlewareInterface $middlew
                     return false;
                 }
 
-                if ($matchedRoute !== $route1) {
+                if ($matchedRoute !== $implicitRoute) {
                     return false;
                 }
```

Comment:
Another change I made in my testing branch was within these argument assertions. Conditionals returning `false` make debugging difficult, as you get a trace that shows a particular mock call was made that was unexpected, but it's very difficult to determine _why_. Instead, use `Assert::{assertion}` in these, and then return `true` or the argument value at the end; when an assertion fails, you find out exactly why, which makes understanding the failures much easier.